fedimint/src/lib.rs

```diff
@@ -178,12 +178,28 @@ impl FedimintServer {
     }
 
     // Build a `ConsensusOutcome` then use the API to validate and process missed epochs
-    pub async fn rejoin_consensus(&mut self) {
+    pub async fn rejoin_consensus(
+        &mut self,
+        timeout: Duration,
+        rng: &mut (impl RngCore + CryptoRng + Clone + 'static),
+    ) {
```

Comment:
Perhaps document what the timeout is for. Just looking at the API I would assume that you only have `timeout`-long to download the entire history. But seems the timeout is only for figuring out which epoch to rejoin at?

Reply:
Sure I'll add some docs.

The timeout is in the case that peers don't agree on a next_epoch, give them all time to respond, but don't get blocked by evil peers.